src/include/types.h

```diff
@@ -534,5 +534,30 @@ WRITE_CLASS_ENCODER(errorcode32_t)
 WRITE_EQ_OPERATORS_1(errorcode32_t, code)
 WRITE_CMP_OPERATORS_1(errorcode32_t, code)
 
+constexpr std::uint64_t operator"" _u64 (unsigned long long n) {
+  return uint64_t(n);
+}
+constexpr std::uint32_t operator"" _u32 (unsigned long long n) {
+  return uint32_t(n);
+}
+constexpr std::uint16_t operator"" _u16 (unsigned long long n) {
+  return uint16_t(n);
+}
+constexpr std::uint8_t operator"" _u8 (unsigned long long n) {
+  return uint8_t(n);
+}
+
+constexpr std::int64_t operator"" _s64 (unsigned long long n) {
+  return int64_t(n);
+}
+constexpr std::int32_t operator"" _s32 (unsigned long long n) {
+  return int32_t(n);
+}
+constexpr std::int16_t operator"" _s16 (unsigned long long n) {
+  return int16_t(n);
+}
+constexpr std::int8_t operator"" _s8 (unsigned long long n) {
+  return int8_t(n);
+}
```

Comment:
user-defined literals are cool, but i think it's overkill for integral types. first, there are already well-known suffixes for these literals (granted they don't map perfectly to these stdint types). more importantly, though, is that the compiler won't warn you about narrowing conversions when you use these. both gcc and clang are perfectly happy to convert `1048576_u16` into `uint16_t(0)` (see https://godbolt.org/g/m7SygL)